includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -1326,13 +1326,54 @@ public function get_products( $args = array() ) {
 	public function search_products( $term, $type = '', $include_variations = false, $all_statuses = false ) {
 		global $wpdb;
 
-		$like_term     = '%' . $wpdb->esc_like( $term ) . '%';
 		$post_types    = $include_variations ? array( 'product', 'product_variation' ) : array( 'product' );
 		$post_statuses = current_user_can( 'edit_private_products' ) ? array( 'private', 'publish' ) : array( 'publish' );
 		$type_join     = '';
 		$type_where    = '';
 		$status_where  = '';
 
+		// See if search term contains OR keywords.
+		if ( stristr( $term, ' OR ' ) ) {
+			$term_groups = explode( ' OR ', $term );
+		} else {
+			$term_groups = array( $term );
+		}
+
+		$search_where   = '';
+		$search_queries = array();
+
+		foreach ( $term_groups as $term_group ) {
+			// Parse search terms.
+			if ( preg_match_all( '/".*?("|$)|((?<=[\t ",+])|^)[^\t ",+]+/', $term_group, $matches ) ) {
+				$search_terms = $this->get_valid_search_terms( $matches[0] );
+				$count        = count( $search_terms );
+
+				// if the search string has only short terms or stopwords, or is 10+ terms long, match it as sentence.
+				if ( 9 < $count || 0 === $count ) {
+					$search_terms = array( $term_group );
+				}
+			} else {
+				$search_terms = array( $term_group );
+			}
+
+			$term_group_query = '';
+			$searchand        = '';
+
+			foreach ( $search_terms as $search_term ) {
+				$like              = '%' . $wpdb->esc_like( $search_term ) . '%';
+				$term_group_query .= $wpdb->prepare( " {$searchand} ( ( posts.post_title LIKE %s) OR ( posts.post_content LIKE %s ) OR ( postmeta.meta_key = '_sku' AND postmeta.meta_value LIKE %s ) )", $like, $like, $like ); // @codingStandardsIgnoreLine.
+				$searchand         = ' AND ';
```

Comment:
Should this be AND? I tested with + , and in both cases it used the AND. My understanding would be that using , should OR this? IE. Searching for `hoodie+pocket,hoodie+logo` should return both the hoodie with logo and hoodie with pocket products, but instead with this it returns nothing?

Reply:
hoodie+pocket or hoodie+logo